src/store/lookup.ts

```diff
@@ -203,6 +204,11 @@ async function lookup(browser: Browser, store: Store) {
 						return onProxyFunc;
 					}
 
+					// Give dummy setRequestInterception to avoid AdBlock from messing with it
+					if (prop === 'setRequestInterception') {
+						return noop;
+					}
```

Comment:
I suppose the important part here is that you're returning an empty function rather than a value, correct?

Why is this important?

Reply:
The adblock plugin will sometimes, for various reasons, call `setRequestInterception(false)` because it thinks it is the authority on request listeners (which, due to us injecting proxies with that, too, it is not).
All this does is give the adblock plugin a function to call that will not error and simply does nothing at all, so it can be happy thinking it disabled request listeners, and the proxy plugin can be happy still using them.

Reply:
Aha, I see! Interesting. Do you think it would be worth mentioning making a ticket about on their repos?

Reply:
@jef Not really. The way puppeteer works, only one on-request listener is possible. The `Proxy` thing we do is a hack, and for sure not an expected use case of anything. Now if puppeteer ever supports multiple on-request listeners, then that might be considered an issue.

Reply:
Wait, I'm sorry. I'm misunderstanding the situtation. You did make a fork of package and you fixed that issue. As for https://github.com/cliqz-oss/adblocker/issues/1495, that isn't an issue for that library, yet an issue with the original `puppeteer-page-proxy`.

I'm I understanding this correct?

Reply:
@jef The issue with adblock purely arises because we need multiple things that employ request listeners, which puppeteer does not support in any way. So we hack around and "fake" the Page object the adblocker sees to be able to do it. Of course adblock has no way of expecting to get a proxied page object, so this is not a bug on their end at all, we just do something puppeteer cannot do.